tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -1387,6 +1437,54 @@ private[optimizer] abstract class OptimizerCore(semantics: Semantics) {
       case IdentityHashCode =>
         contTree(CallHelper("systemIdentityHashCode", newArgs)(IntType))
 
+      // scala.runtime.ScalaRunTime object
+
+      case ArrayApply =>
+        val List(array, index) = newArgs
+        array.tpe match {
+          case arrayTpe @ ArrayType(base, depth) =>
+            val elemType = cursoryArrayElemType(arrayTpe)
+            val select = ArraySelect(array, index)(elemType)
+            if (base == "C")
+              boxChar(select)(cont)
+            else
+              contTree(select)
+
+          case _ =>
+            defaultApply("array$undapply__O__I__O", AnyType)
+        }
+
+      case ArrayUpdate =>
+        val List(tarray, tindex, tvalue) = targs
+        tarray.tpe.base match {
+          case arrayTpe @ ArrayType(base, depth) =>
+            val array = finishTransformExpr(tarray)
+            val index = finishTransformExpr(tindex)
+            val elemType = cursoryArrayElemType(arrayTpe)
+            val select = ArraySelect(array, index)(elemType)
+            val cont1: PreTransCont = { tunboxedValue =>
+              contTree(Assign(select, finishTransformExpr(tunboxedValue)))
+            }
+            base match {
+              case "Z" | "B" | "S" | "I" | "L" | "F" | "D" if depth == 1 =>
+                foldUnbox(tvalue, base.charAt(0))(cont1)
+              case "C" if depth == 1 =>
+                unboxChar(tvalue)(cont1)
+              case _ =>
+                cont1(tvalue)
+            }
+          case _ =>
+            defaultApply("array$undupdate__O__I__O__V", AnyType)
+        }
+
+      case ArrayLength =>
+        targs.head.tpe.base match {
+          case _: ArrayType =>
+            contTree(Trees.ArrayLength(newArgs.head))
```

Comment:
Yes, because there is a clash with the constant for the intrinsic `ArrayLength`. Intrinsic constants are not imported often, but here they are and it clashes.